includes/Classifai/Providers/OpenAI/Moderation.php

```diff
@@ -0,0 +1,337 @@
+			'moderate' === $action &&
+			$comment_id &&
+			wp_verify_nonce( $nonce, 'moderate_comment' )
+		) {
+			$this->moderate_comment( $comment_id );
+			wp_redirect( '/wp-admin/edit-comments.php' );
+			exit;
+		}
+	}
+
+	/**
+	 * Add action to comment row
+	 *
+	 * @param array       $actions Comment row action
+	 * @param \WP_Comment $comment Comment object
+	 * @return mixed
+	 */
+	public function comment_row_actions( $actions, $comment ) {
+		$nonce = wp_create_nonce( 'moderate_comment' );
+
+		$actions['moderate'] = sprintf(
+			'<a href="%s" aria-label="%s">%s</a>',
+			"edit-comments.php?a=moderate&amp;c={$comment->comment_ID}&amp;nonce={$nonce}",
+			esc_attr__( 'Moderate this comment' ),
+			__( 'Moderate' )
+		);
+
+		return $actions;
+	}
+
+	/**
+	 * Check to see if the feature is enabled and a user has access.
+	 *
+	 * @return bool|WP_Error
+	 */
+	public function is_feature_enabled() {
+		$settings = $this->get_settings();
+
+		// Check if valid authentication is in place.
+		if ( empty( $settings ) || ( isset( $settings['authenticated'] ) && false === $settings['authenticated'] ) ) {
+			return new WP_Error( 'auth', esc_html__( 'Please set up valid authentication with OpenAI.', 'classifai' ) );
+		}
+
+		// Check if the current user has permission.
+		$roles      = $settings['roles'] ?? [];
+		$user_roles = wp_get_current_user()->roles ?? [];
+
+		if ( empty( $roles ) || ! empty( array_diff( $user_roles, $roles ) ) ) {
+			return new WP_Error( 'no_permission', esc_html__( 'User role does not have permission.', 'classifai' ) );
+		}
+
+		if ( ! current_user_can( 'moderate_comments' ) ) {
+			return new WP_Error( 'no_permission', esc_html__( 'User does not have permission to moderate comments.', 'classifai' ) );
+		}
+
+		// Ensure feature is turned on.
+		if ( ! isset( $settings['enable_moderation'] ) || 1 !== (int) $settings['enable_moderation'] ) {
+			return new WP_Error( 'not_enabled', esc_html__( 'Comment moderation is not enabled.', 'classifai' ) );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Start the audio transcription process.
+	 *
+	 * @param int $comment_id Attachment ID to process.
+	 * @return void
+	 */
+	public function moderate_comment( $comment_id = 0 ) {
+		$settings = $this->get_settings();
+		$enabled  = $this->is_feature_enabled();
+
+		if ( is_wp_error( $enabled ) ) {
+			return;
+		}
+
+		$api_key = $settings['api_key'];
+		$comment = get_comment( $comment_id );
+
+		$api_response = wp_remote_post(
+			'https://api.openai.com/v1/moderations',
+			[
+				'headers' => [
+					'Content-Type'  => 'application/json',
+					'Authorization' => 'Bearer ' . $api_key,
+				],
+				'body'    => wp_json_encode(
+					[
+						'input' => $comment->comment_content,
+					]
+				),
+			]
+		);
+
+		if ( ! is_wp_error( $api_response ) && 200 === wp_remote_retrieve_response_code( $api_response ) ) {
+			$body    = json_decode( wp_remote_retrieve_body( $api_response ), true );
+			$flagged = $body['results'][0]['flagged'];
```

Comment:
Are we certain the response will always contain array up to 3 levels of nesting, or should we perform a check using `isset()` just to be safe? Ignore this comment if we are sure the response format will always be consistent.

Reply:
The response will be consistent :)